bfa_rust/lib/z3floats.ml

```diff
@@ -0,0 +1,97 @@
+open Simple_smt
+
+(* Helpful: https://smt-lib.org/theories-FloatingPoint.shtml *)
+
+let rm = atom "RNA" (* equivalent to roundNearestTiesToAway *)
+let t_f16 = atom "Float16"
+let t_f32 = atom "Float32"
+let t_f64 = atom "Float64"
+let t_f128 = atom "Float128"
+
+let f32_k f =
+  let bin = Int32.bits_of_float f in
+  (* a Float32 has 8 exponent bits, 23 explicit mantissa bits *)
+  app_ "fp"
+    [
+      bv_nat_bin 1 (if Float.sign_bit f then Z.one else Z.zero);
+      bv_nat_bin 8 (Z.of_int32 @@ Int32.shift_right_logical bin 23);
+      bv_nat_bin 23 (Z.of_int32 @@ Int32.logand bin 0x7fffffl);
+    ]
```

Comment:
Imma be honest, no idea what's happening here lmao
Could you just drop a small document in the Obsidian?